### PR TITLE
Throw an error if we fail to access Theme API

### LINKED ISSUE #428: `npm start` fails silently if the API is set to NO ACCESS
### Problem
While developing a theme:

Running `npm start` fails silently if the user hasn't granted access:

image

We should show an error stating that the user needs to grant access.

### Replication steps

- Set API access to `no access`
- Run `npm start`

@t-kelly

### PR DESCRIPTION
**Pull request by t-kelly**
### What are you trying to accomplish with this PR?

Fixes #428 

We were not catching any request errors when checking if we were deploying to the store's published theme. This PR makes sure we are handling any errors emitted from this action.

### CODE REVIEW

File: packages/slate-tools/cli/commands/start.js [hunk 1]
```diff
@@ -23,7 +23,7 @@ const options = {
 const spinner = ora(chalk.magenta(' Compiling...'));
 let firstSync = true;
 let skipSettingsData = null;
-let continueIfPublishedTheme = null;
+const continueIfPublishedTheme = null;
```
Comment (chrisberthe) [inline packages/slate-tools/cli/commands/start.js]: Should be `let`.

File: packages/slate-tools/cli/commands/start.js [hunk 2]
```diff
@@ -81,13 +81,22 @@ devServer.compiler.hooks.done.tap('CLI', (stats) => {
 });
 
 devServer.client.hooks.beforeSync.tapPromise('CLI', async (files) => {
+  console.log('sync');
```
Comment (chrisberthe) [inline packages/slate-tools/cli/commands/start.js]: We can remove this.

File: packages/slate-sync/index.js [hunk 3]
```diff
@@ -52,6 +53,7 @@ function _generateIgnoreFlags() {
  * @return          Promise
  */
 async function deploy(cmd = '', files = []) {
+  console.log('deploy start');
```
Comment (chrisberthe) [inline packages/slate-sync/index.js]: We can remove this.

File: packages/slate-sync/index.js [hunk 4]
```diff
@@ -119,6 +121,70 @@ function promiseThemekitDeploy(cmd, files) {
...
+          if (!Array.isArray(parsed.themes)) {
+            reject(`
+              Shopify response for /admin/themes.json is not an array.
+
+              ${JSON.stringify(parsed, null, '\t')}
+            `);
+            return;
+          }
+
+          const mainTheme = parsed.themes.find((t) => t.role === 'main');
```
Comment (chrisberthe) [inline packages/slate-sync/index.js]: Out of curiosity, are there ever any instances where the published theme doesn't have a role of "main"?

Comment (t-kelly) [reply packages/slate-sync/index.js]: Hmmm... @davidcornu? 👋^^

Comment (NathanPJF) [reply packages/slate-sync/index.js]: @chrisberthe yep, for legacy reasons a published theme can have a role of `main` or `mobile`.  There was once a time when Shopify supported two published themes - one for desktop and one for mobile. Mobile has long been deprecated, but may exist in the wild.  So you want to make sure you're pushing to the `main` one always.

Comment (chrisberthe) [reply packages/slate-sync/index.js]: Interesting! Makes sense, thanks for the clarification.

**Pull request merged**